adi/adar1000.py

```diff
@@ -100,7 +100,8 @@ def __init__(self, uri="", beams="BEAM0"):
                 for dev in self._ctx.devices:
                     if "label" in dev.attrs and dev.attrs["label"].value == beam:
                         self._ctrls.append(dev)
-            if len(self._ctrls) == len(beams):
+                        break
```

Comment:
The loop doesn't need to keep checking devices if we've already found the right one.